Week1/app.js

```diff
@@ -0,0 +1,205 @@
+'use strict';
+function main() {
+  //making the main div
+  let mainDiv = document.createElement('div');
+  document.body.appendChild(mainDiv);
+
+  //giving an id to main div
+  let mainDivAttribute = document.createAttribute('id');
+  mainDivAttribute.value = 'root';
+  mainDiv.setAttributeNode(mainDivAttribute);
+
+  function createHTMLElement(tag, parent) {
+    let newElement = document.createElement(tag);
+    if (parent) {
+      parent.appendChild(newElement);
+    }
+    else {
+      document.body.appendChild(newElement);
+    }
+    return newElement;
+  }
+
+
+  let header = createHTMLElement('header', mainDiv);
+
+  let title = createHTMLElement('h1', header);
+  title.innerText = 'Hack Your Future repositories';
+
+  let repositoryInformationBox = createHTMLElement('div', mainDiv);
+  repositoryInformationBox.setAttribute('id', 'rep_div');
+
+
+  //Requesting HYF repositories
+  const user = 'HackYourFuture';
+  const gitHubUrl = 'https://api.github.com/users/' + user + '/repos?per_page=100';
+  const hyfRepositoriesRequest = new XMLHttpRequest();
+  hyfRepositoriesRequest.open('GET', gitHubUrl, true);
+  hyfRepositoriesRequest.send();
+  hyfRepositoriesRequest.onreadystatechange = repositoriesRequest;
```

Comment:
You have not handled any errors if server is not available. This way the user doesn't know why the website is not working when a server is not available.